lang/java/avro/src/main/java/org/apache/avro/file/DataFileWriter.java

```diff
@@ -234,6 +235,28 @@ public boolean isFlushOnEveryBlock() {
     return this;
   }
 
+  /**
```

Comment:
` public DataFileWriter<D> appendTo(SeekableInput in, OutputStream out) ` could delegate to the new `public DataFileWriter<D> appendTo(Header header, OutputStream out)` after it gets the `header` from the `reader`.